### PR TITLE
[DUOS-2475] Add Elasticsearch API and search page

### PR DESCRIPTION
**Pull request by fboulnois**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2475

### Summary

Add Elasticsearch API and search page

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/pages/DatasetSearch.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                                "match": {
+                                    "_id": 1403
+                                }
+                            }
+                        ]
+                    }
+                }
+            };
+            try {
+                const res = await DataSet.searchDatasetIndex(example);
```
Comment (fboulnois) [inline src/pages/DatasetSearch.js]: I initially created a component for this data as per the ticket, but it didn't seem like it was very useful.

**Review by rushtong — COMMENTED** [review]
I agree with the codacy suggestions here, otherwise it looks good 👍🏽

**Review by hams7504 — COMMENTED** [review]
Other than the Codacy errors, lgtm 👍

**Review by rushtong — APPROVED** [review]
👍🏽

**Review by hams7504 — APPROVED** [review]
👍

**Pull request merged**